dbms/Common/HashTable/StringHashMap.h

```diff
@@ -25,9 +25,13 @@ struct StringHashMapCell<StringKey16, TMapped> : public HashMapCell<StringKey16,
     using Base::Base;
     static constexpr bool need_zero_value_storage = false;
     bool isZero(const HashTableNoState & state) const { return isZero(this->value.first, state); }
-    // Assuming String does not contain zero bytes. NOTE: Cannot be used in serialized method
-    static bool isZero(const StringKey16 & key, const HashTableNoState & /*state*/) { return key.low == 0; }
-    void setZero() { this->value.first.low = 0; }
+
+    // Zero means unoccupied cells in hash table. Use key with last word = 0 as
+    // zero keys, because such keys are unrepresentable (no way to encode length).
+    static bool isZero(const StringKey16 & key, const HashTableNoState &)
```

Comment:
Why we only check high half?
It is because we store length in high half?

Reply:
yep